### PR TITLE
Adds STA Emulation

### PR DESCRIPTION
**Pull request by rafaeldelboni**

### CODE REVIEW

File: src/cpu.c [hunk 1]
```diff
@@ -23,9 +25,143 @@ void resetCpu(struct CPU * cpu, struct BUS * bus) {
   cpu->status.negative = false;
 }
 
+unsigned int absoluteAddressMode(struct CPU *cpu, struct BUS *bus) {
+  unsigned char lo = readBus(cpu->pc, bus);
```
Comment (lucascebertin) [inline src/cpu.c:29]: Any chance to encapsulate absolute behaviour?
They are pretty similar

Comment (lucascebertin) [reply src/cpu.c:29]: I mean, all those absolute* functions

File: src/cpu.c [hunk 2]
```diff
@@ -23,9 +25,143 @@ void resetCpu(struct CPU * cpu, struct BUS * bus) {
...
+  cpu->pc++;
+  unsigned int base = readBus((base_hi << 8) | base_lo, bus);
+  unsigned char lo = readBus(base, bus);
+  unsigned char hi = readBus(base+1, bus);
+  unsigned int address = (hi << 8) | lo;
+  return address;
+}
+
+unsigned int indirectXAddressMode(struct CPU *cpu, struct BUS *bus) {
+  unsigned int base = readBus(cpu->pc, bus) + cpu->index_x;
```
Comment (lucascebertin) [inline src/cpu.c:59]: Same as previous comment, can we encapsulate indirect functions?

File: src/cpu.c [hunk 3]
```diff
@@ -23,9 +25,143 @@ void resetCpu(struct CPU * cpu, struct BUS * bus) {
...
+}
+
+unsigned int relativeAddressMode(struct CPU *cpu, struct BUS *bus) {
+  signed char address = cpu->pc + readBus(cpu->pc, bus);
+  cpu->pc++;
+  return address;
+}
+
+unsigned int zeroPageAddressMode(struct CPU *cpu, struct BUS *bus) {
+  unsigned int address = readBus(cpu->pc, bus);
```
Comment (lucascebertin) [inline src/cpu.c]: Same as previous comment, can we encapsulate zero page functions?

File: test/cpu_test.c [hunk 4]
```diff
@@ -29,12 +29,145 @@ void test_function_should_reset_cpu(void) {
...
+  cpu.accumulator = 0xDD; 
+
+  /* STA $0d */
+  writeBus(0x600, 0x85, &bus);
+  writeBus(0x601, 0x0D, &bus);
+
+  clockCpu(&cpu, &bus);
+
+  // should move pc 2 times
+  TEST_ASSERT_EQUAL(0x602, cpu.pc);
```
Comment (lucascebertin) [inline test/cpu_test.c:43]: NIIIIIIIIIIIIIIIIIIICE

File: test/cpu_test.c [hunk 5]
```diff
@@ -29,12 +29,145 @@ void test_function_should_reset_cpu(void) {
...
+  TEST_ASSERT_EQUAL(cpu.accumulator, readBus(0x4400, &bus));
+}
+
+void test_cpu_should_sta_absolute_x(void) {
+  // STA Absolute X with pre loaded register: Accumulator=0xDD and X=0x02
+  cpu.accumulator = 0xDD;
+  cpu.index_x = 0x02;
+
+  /* STA $4400,X */
+  writeBus(0x600, 0x9d, &bus);
```
Comment (lucascebertin) [inline test/cpu_test.c:88]: Write bus receives offset and hex opcode respectively, right?
Naice!

**Review by lucascebertin — APPROVED** [review]

File: src/cpu.c (hunk 2 above)
Comment (rafaeldelboni) [reply src/cpu.c:59]: Indirect have some different behaviors between them, I extracted some redundant code anyways.

**Pull request merged**